File: app/src/main/java/com/playgilround/schedule/client/signup/SignUpContract.java
package com.playgilround.schedule.client.signup;

import com.playgilround.schedule.client.base.BasePresenter;
import com.playgilround.schedule.client.base.BaseView;

public interface SignUpContract {

    interface View extends BaseView<Presenter> {

        void fieldCheck(boolean check);

        void signUpError(int status);

        void singUpComplete();

    }

    interface Presenter extends BasePresenter {

        void onClickNext(int position);

        void onClickBack(int position);

        void signUp();

    }
}

File: app/src/main/java/com/playgilround/schedule/client/signup/SignUpContract.kt
package com.playgilround.schedule.client.signup

import com.playgilround.schedule.client.base.BasePresenter
import com.playgilround.schedule.client.base.BaseView

interface SignUpContract {

    interface View: BaseView<Presenter> {
        fun fieldCheck(check: Boolean)

        fun signUpError(status: Int)

        fun signUpComplete()
    }

    interface Presenter: BasePresenter {
        fun onClickNext(position: Int)

        fun onClickBack(position: Int)

        fun validateEmail(email: String): Boolean

        fun validateNickName(nickName: String): Boolean

        fun signUp()
    }
}